content/chat/rooms/moderation/hive-model-only.textile

```diff
@@ -9,4 +9,29 @@ languages:
   - kotlin
 ---
 
-Hive documentation to go here.
+"Hive Moderation":https://hivemoderation.com is a powerful suite of moderation tools that can be used to moderate content in chat rooms.
+
+The Hive (Model Only) rule is a rule applied to chat rooms in Ably Chat which allows you to use Hive's text moderation models to detect and handle inappropriate content before it is published to other users.
+
+h2(#fields). Rule fields
+
+- Hive API key := The API key for your Hive account.
+- Thresholds := A map of text "text moderation classes":https://docs.thehive.ai/reference/text-moderation to "severity":https://docs.thehive.ai/docs/detailed-class-descriptions-text-moderation. When moderating text, any message deemed to be at or above a specified threshold will be rejected and not published to the chat room.
+- Model URL (optional) := A custom URL if using a custom moderation model.
+- Retry timeout := Maximum duration (in milliseconds) that an attempt to invoke the rule may take (including any retries).
+- Max retries := Maximum number of retries after the first attempt at invoking the rule.
+- Failed action := The action to take in the event that a rule fails to invoke. Options are reject the request or publish anyway.
+- Too many requests action := The action to take in the event that Hive returns a 429 (Too Many Requests Response). Options are to fail rule invocation, or retry.
+- Room filter (optional) := A regular expression to match to specific chat rooms.
+
+h2(#chunking). Content chunking
```

Comment:
Is it possible to reword this? I'm not sure chunking is reading right with me. I would guess it's more along the lines of Text limits and then maybe something like:

```
Hive's models accept content with a maximum length of 1024 characters. If sending a message larger than this, Ably will automatically break the text into smaller requests, with crossover between segments to ensure context is preserved.
```

What do you think?

Reply:
Yes agreed, and added a section at the bottom re how we treat the chunks